airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/strategies/cursor_pagination_strategy.py

```diff
@@ -11,57 +13,122 @@
 from airbyte_cdk.sources.declarative.interpolation.interpolated_boolean import InterpolatedBoolean
 from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
 from airbyte_cdk.sources.declarative.requesters.paginators.strategies.pagination_strategy import PaginationStrategy
-from airbyte_cdk.sources.declarative.types import Config
+from airbyte_cdk.sources.declarative.types import Config, Record
+
+DECODED_RESPONSE_TYPE = Union[Mapping[str, Any], List[dict[str, Any]]]
```

Comment:
this is needed because the decoder can return either a mapping or a list, is this correct? If so, can you update the decoder to return a value of this type to document why this is needed